### DISCUSSION TITLE
Temporary lock multiple parts relatively to a robot body

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#4327] [Q&A]
I have a mobile robot with a complex arm, and the simulation of the set of joint of full arm cost a lot in term of performance. Mainly during phases during which the robot drive itself from two location, during which the arm don't move relatively to the main body of the robot.

It would be nice if we could temporary "disable" the ODE computation on (the joint between) those part to save time, and reactivate them only when the arm actually move. The idea is to kinda transform a complex arm into an object that behave like a simple solid (with updated CoM, inertia...)

I do think I'm able to achieve it with a lot of Connector, that, as far as my experience with them goes, disable constraints between the connected part. But that would make me use a lot of connectors that would be very artificial and not pretty...

Is there any possibility that I could have missed to achieve that ?

**Comment by omichel** [comment]
You could also simply have a special PROTO of the arm with a boolean "physics" parameter. If that parameter is TRUE, it would be like a normal arm. If it is FALSE, it would generate a plain solid node with no joints inside instead of a multi-join structure. Changing this parameter from TRUE to FALSE from a supervisor process would remove the physics of your robot by regenerating the PROTO. Changing this parameter from FALSE to TRUE would re-enable the physics.

**Reply by ShuffleWire** [reply]
that's interesting ! I have to figure out a way to be able to switch back and forth smoothly, while keeping mass and CoM reliable and representative of the current position of the arm...
I will try to play with it.
Does Lua is the way to go ?

**Reply by ad-daniel** [reply]
preferably JavaScript as Lua will be deprecated at some point, but both work at the moment.
